File: Dataset/Leetcode/train/104/466.py
class Solution:
    def XXX(self, root: TreeNode) -> int:
        if not root:
            return 0
        def zcc(left,right):
            if not (left or right):
                return 0
            elif not left:
                return 1 + zcc(right.left,right.right)
            elif not right:
                return 1 + zcc(left.left,left.right)
            return 1 + max(zcc(left.left,left.right),zcc(right.left,right.right))
        return zcc(root.left,root.right)+1

File: Dataset/Leetcode/train/111/466.java
class Solution {
    public int XXX(TreeNode root) {
        
        return height(root);
    }

    public int height(TreeNode node) {
        if (node == null) {
            return 0;
        }
        if (node.left == null && node.right == null) {
            return 1;
        }

        int leftHeight = height(node.left) + 1;

        int rightHeight = height(node.right) + 1;

        if (node.left == null || node.right == null) {
            return node.left == null ? rightHeight : leftHeight;
        }

        return Math.min(leftHeight, rightHeight);
    }
}

